### DISCUSSION TITLE
Switiching tariff between Agile and Intelligent with the seasons

### DISCUSSION THREAD

**Discussion by ragg987** [BottlecapDave/HomeAssistant-OctopusEnergy#938] [Q&A]
I've been using this add-on with my Agile tariff for about 7-8 months, and have managed to create a number of automations using e.g. Target Rate sensors to control my space heating, hot water, appliances, EV charging...

Thanks to a post by @plord12 in the myenergy add-on I now intend to switch my tariff to Intelligent Go + Outgoing for Export so that I can generate some income in the summer months. My thinking is to use Intelligent in the warmer 7-8 months when I don't need heating and generate quite a lot of solar, then switch to Agile in the colder months.

Can anyone suggest how I can cater for 2 sets of automations, one set for Intelligent and another for Agile?

Some thoughts:

- add a condition into my automation to check for agile or intelligent? Would require editing every automation. How can I detect which tariff I am on?
- do something with area, labels or categories - I don't know if it is possible to e.g. enable or disable all automations in a category?
- other options?

Any experiences or input would be welcome...

**Comment by ragg987** [comment]
Partial answer to above: "how do I detect which tariff I am on"
```
"{{ 'AGILE' in state_attr('sensor.electricity_XXX_YYY_current_rate','tariff') }}"
```
results in "True" with current Agile tariff

**Comment by BottlecapDave** [comment]
Hello and sorry for the late response. You could potentially use the target rate sensors to create binary sensors that turn on for a target number of hours. These sensors use your current tariff and available rates which means they'd automatically adapt when you switch tariffs. For example if you set it up to look at 4pm-4pm for 1 hour, it would turn on during the night when on intelligent but might turn on during the day when on agile (depending on the rates).

**Comment by ragg987** [comment]
Hi no worries about the delay in response - I appreciate you have taken the time to get back.

I already use target rate sensors quite a lot and they may be a partial solution, however in most cases I don't want to give the target such a wide range of times (24h in your example) and want tighter control. I have decided to proceed with 2 sets of automations with a condition within the automation that checks which tariff is current. I am using a template helper that tracks the current tariff to support this. The helper looks like this (untested for IOG yet as I'm still on Agile, awaiting Octopus to register my export meter before I switch to IOG).
```
{% set tariff_code = state_attr('sensor.electricity_XXX_YYY_current_rate','tariff') %}
{% if 'AGILE' in tariff_code %}
  octopus_agile
{% elif 'INTELLIGENT' in tariff_code %}
  octopus_intelligent
{% else %}
  tariff_error
{% endif %}
```
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/127891408/4786022f-5d0a-40ce-afd1-d90dac18ccbb)

I was hoping someone could share a snazzy method, I have seen a number of people mention they switch tariffs between seasons.

**Reply by mikelward** [reply]
I find it easiest to have one automation per device. Then it's clear where it's defined. And I can define common things like triggers, fields, and variables only once. 

I'd have a template sensor, like I think you're doing. Then the top level sequence action would be a choose, with an option for each tariff.

I haven't tried this for switching tariffs, but I do something similar to control my batteries.
